akka-http-core/src/main/scala/akka/http/scaladsl/model/RequestResponseAssociation.scala

```diff
@@ -16,4 +17,14 @@ trait RequestResponseAssociation
  * A simple value holder class implementing RequestResponseAssociation.
  */
 @ApiMayChange
-final case class SimpleRequestResponseAttribute[T](value: T) extends RequestResponseAssociation
+case class SimpleRequestResponseAttribute[T](value: T) extends RequestResponseAssociation
+
+/**
+ * An association for completing a future when the response arrives.
+ */
+final class ResponseFuture(val promise: Promise[HttpResponse]) extends RequestResponseAssociation
```

Comment:
Maybe `ResponsePromise` or `ResponseCallback`, instead?

Reply:
Heh, I agree, I had it as `ResponsePromise` but couldn't get the `@apidoc` paradox tag to work with two different names. The best reason for naming things. Perhaps I'll just skip those links.